docs/source/user_guide/basic_concepts/expressions.rst

```diff
@@ -314,3 +314,188 @@ The :meth:`.if_else() <daft.expressions.Expression.if_else>` method is a useful
+    +------------------------+------------------------+-------------------------------+
+    | 2021-01-01T00:02:00    | 2021-01-01T00:02:10    | 0:00:10                       |
+    +------------------------+------------------------+-------------------------------+
+    (Showing first 3 rows)
+
+
+Extracting Time Stamps
+^^^^^^^^^^^^^^^^^^^^^^
+
+You can extract parts of a Timestamp, such as the year, month, day, hour, minute, and second.
+
+.. code:: python
+
+    df = df.with_columns("year", df["x"].dt.year()) // with columns integration
+    df = df.with_column("month", df["x"].dt.month())
+    df = df.with_column("day", df["x"].dt.day())
+    df = df.with_column("hour", df["x"].dt.hour())
+    df = df.with_column("minute", df["x"].dt.minute())
+    df = df.with_column("second", df["x"].dt.second())
+    df.show()
+
+.. code:: none
+
+    +------------------------+------+------+-----+------+-------+--------+
+    | x                      | year | month| day | hour | minute| second |
+    | DateTime               | Int32| Int32|Int32| Int32| Int32 | Int32  |
+    +========================+======+======+=====+======+=======+========+
+    | 2021-01-01T00:01:01    | 2021 | 1    | 1   | 0    | 1     | 1      |
+    +------------------------+------+------+-----+------+-------+--------+
+    | 2021-01-01T00:01:59    | 2021 | 1    | 1   | 0    | 1     | 59     |
+    +------------------------+------+------+-----+------+-------+--------+
+    | 2021-01-01T00:02:00    | 2021 | 1    | 1   | 0    | 2     | 0      |
+    +------------------------+------+------+-----+------+-------+--------+
+    (Showing first 3 rows)
+
+
+Converting Between Time Zones
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can convert a Timestamp to a different time zone.
+
+.. code:: python
+
+    df = daft.from_pydict({
+        "json": [
+            '{"a": 1, "b": 2}',
+            '{"a": 3, "b": 4}',
+        ],
+    })
+    df = df.with_column("a", df["json"].json.query(".a"))
+    df.collect()
+
+.. code:: none
+
+    ╭──────────────────┬──────╮
+    │ json             ┆ a    │
+    │ ---              ┆ ---  │
+    │ Utf8             ┆ Utf8 │
+    ╞══════════════════╪══════╡
+    │ {"a": 1, "b": 2} ┆ 1    │
+    ├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌┤
+    │ {"a": 3, "b": 4} ┆ 3    │
+    ╰──────────────────┴──────╯
+
+    (Showing first 2 of 2 rows)
+
+
+Using the Truncate Function
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The `truncate` function can be used to truncate dates to a specific time unit. This can be useful when you want to group or round dates to the nearest week, month, etc.
+
+.. code:: python
+
+    import pandas as pd
+    import daft
+
+    # Create a DataFrame with a range of dates
+    df = daft.from_pydict({
+        "dates": pd.date_range(start="2023-07-01", end="2023-07-06", freq='D').to_list()
+    })
+
+    # Truncate dates to the start of the week
+    df = df.with_column("truncated_dates", df["dates"].dt.truncate(freq="W-SUN"))
```

Comment:
Does this code work? I'm pretty sure the truncate method doesn't expose a `freq` argument